engine/src/hndlrlst.cpp

```diff
@@ -659,3 +663,150 @@ bool MCHandlerlist::enumerate(MCExecPoint& ep, bool p_first)
 	
 	return p_first;
 }
+
+// SN-2015-05-28: [[ Bug 11277 ]] Similiar function to MCHandler::deletestatements
+void MCHandlerlist::deletestatements(MCStatement *p_statements)
+{
+    while (p_statements != NULL)
+    {
+        MCStatement *tsptr = p_statements;
+        p_statements = p_statements->getnext();
+        delete tsptr;
+    }
+}
+
+Exec_stat MCHandlerlist::eval(MCExecPoint& ep)
+{
+#ifdef /* MCHandlerlist */ LEGACY_EXEC
+    // SN-2015-05-28: [[ Bug 11277 ]] Implement the parsing of the expression -
+    //  same as MCHandler::eval, but no handler is set the ScriptPoint.
+    MCScriptPoint sp(ep);
+    MCExpression *exp = NULL;
+    Symbol_type type;
+    Exec_stat stat = ES_ERROR;
+    if (sp.parseexp(False, True, &exp) == PS_NORMAL && sp.next(type) == PS_EOF)
+    {
+        stat = exp->eval(ep);
+        ep.grabsvalue();
+    }
+    delete exp;
+    return stat;
+#endif /* MCHandlerlist */
+}
+
+Exec_stat MCHandlerlist::doscript(MCExecPoint& ep, uint2 line, uint2 pos)
```

Comment:
Can we unify MCHandler::doscript and MCHandlerlist::doscript by refactoring them (and the related functions) into MCExecPoint?

This seems an eminently sensible place for them to go as the EP contains all the info it needs to execute in the appropriate context (I think the same is true of eval).